SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/SpatialViewUtils.h

```diff
@@ -0,0 +1,168 @@
+#include "SpatialView/ViewDelta.h"
+#include "SpatialView/WorkerView.h"
+
+namespace SpatialGDK
+{
+enum EntityStatus
+{
+	ADD,
+	REMOVE,
+	UPDATE
+};
+
+inline void AddComponentAddedChange(TArray<ComponentChange>& Changes, int ComponentId, double Value)
+{
+	const auto Data = Schema_CreateComponentData();
+	Schema_Object* Fields = Schema_GetComponentDataFields(Data);
+	Schema_AddDouble(Fields, ComponentId, Value);
+	Changes.Push(ComponentChange(ComponentId, Schema_CopyComponentData(Data)));
+}
+
+inline void AddComponentUpdateChange(TArray<ComponentChange>& Changes, int ComponentId, double Value)
+{
+	const auto Update = Schema_CreateComponentUpdate();
+	Schema_Object* Fields = Schema_GetComponentUpdateFields(Update);
+	Schema_AddDouble(Fields, ComponentId, Value);
+	Changes.Push(ComponentChange(ComponentId, Schema_CopyComponentUpdate(Update)));
+}
+
+inline void AddComponentRemovedChange(TArray<ComponentChange>& Changes, int ComponentId)
+{
+	Changes.Push(ComponentChange(ComponentId));
+}
+
+inline EntityDelta CreateEntityDelta(int EntityId, EntityStatus Status)
+{
+	EntityDelta Delta;
+	Delta.EntityId = EntityId;
+	Delta.bAdded = Status == ADD;
+	Delta.bRemoved = Status == REMOVE;
+	Delta.AuthorityGained = ComponentSpan<AuthorityChange>();
+	Delta.AuthorityLost = ComponentSpan<AuthorityChange>();
+	Delta.AuthorityLostTemporarily = ComponentSpan<AuthorityChange>();
+	Delta.ComponentsAdded = ComponentSpan<ComponentChange>();
+	Delta.ComponentsRemoved = ComponentSpan<ComponentChange>();
+	Delta.ComponentsRefreshed = ComponentSpan<ComponentChange>();
+	Delta.ComponentUpdates = ComponentSpan<ComponentChange>();
+	return Delta;
+}
+
+inline void AddEntityToView(EntityView& View, int EntityId)
+{
+	View.Add(EntityId, { {}, {} });
+}
+
+inline void AddComponentToView(EntityView& View, int EntityId, int ComponentId, double Value)
+{
+	ComponentData TestComponentData = CreateTestComponentData(ComponentId, Value);
+	View[EntityId].Components.Push(MoveTemp(TestComponentData));
+}
+
+inline void AddAuthorityToView(EntityView& View, int EntityId, int ComponentId)
+{
+	View[EntityId].Authority.Push(ComponentId);
+}
+
+inline bool AreEquivalent(EntityView& Lhs, EntityView& Rhs)
+{
+	TArray<Worker_EntityId_Key> LhsKeys;
+	TArray<Worker_EntityId_Key> RhsKeys;
+	Lhs.GetKeys(LhsKeys);
+	Rhs.GetKeys(RhsKeys);
+	if (!AreEquivalent(LhsKeys, RhsKeys, CompareWorkerEntityIdKey))
+	{
+		return false;
+	}
+
+	for (const auto& Pair : Lhs)
+	{
+		const long EntityId = Pair.Key;
+		const EntityViewElement* LhsElement = &Pair.Value;
+		const EntityViewElement* RhsElement = &Rhs[EntityId];
+		if (!AreEquivalent(LhsElement->Components, RhsElement->Components))
+		{
+			return false;
+		}
+
+		if (!AreEquivalent(LhsElement->Authority, RhsElement->Authority, CompareWorkerComponentId))
+		{
+			return false;
+		}
+	}
+
+	return true;
+}
+
+inline bool AreEquivalent(ViewDelta& Lhs, ViewDelta& Rhs)
+{
+	// This does not check for events or commands atm
```

Comment:
nit: is this going to be followed up? If so, can it have a ticket and a TODO, otherwise would rather it was just documented that this method does not compare events and commands